modules/mux/targets/host/source/command_buffer.cpp

```diff
@@ -1189,6 +1189,9 @@ mux_result_t hostResetCommandBuffer(mux_command_buffer_t command_buffer) {
   const std::lock_guard<std::mutex> lock(host->mutex);
 
   host->commands.clear();
+  host->commands.shrink_to_fit();
+  host->ndranges.clear();
+  host->ndranges.shrink_to_fit();
```

Comment:
Why does it need this clear/shrink/clear/shrink? Maybe add a comment as to why?

Reply:
`clear()` just sets the vector's size to zero while leaving its capacity as it was, `shrink_to_fit()` tries to reduce the vector's capacity (the number of elements the vector has reserved memory for) too. Although `host->commands` and `host->ndranges` are defined as cargo's `small_vector`, the API is the same as for `std::vector` ([`clear()`](https://en.cppreference.com/w/cpp/container/vector/clear), [`shrink_to_fit()`](https://en.cppreference.com/w/cpp/container/vector/shrink_to_fit)), so this should hopefully be familiar to people already.

I did expect this would be commented in the Doxygen comments for [`small_vector::clear()`](https://github.com/codeplaysoftware/oneapi-construction-kit/blob/bed85cff4d1f5133a814982e948e22d8dd558c9f/modules/cargo/include/cargo/small_vector.h#L471) and [`small_vector::shrink_to_fit()`](https://github.com/codeplaysoftware/oneapi-construction-kit/blob/bed85cff4d1f5133a814982e948e22d8dd558c9f/modules/cargo/include/cargo/small_vector.h#L459), but it isn't. I'm happy to add a bit more there.

Reply:
It wasn't the api usage as such, but why you do it twice.

Reply:
sorry I misread it the first time, it's fine.